File: java/src/test/java/com/dddeurope/niuz/services/PaymentServiceTest.java
package com.dddeurope.niuz.services;

import com.dddeurope.niuz.dtos.PaymentDto;
import com.dddeurope.niuz.entities.Payment;
import com.dddeurope.niuz.repositories.PaymentRepository;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.util.Arrays;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

public class PaymentServiceTest {
    private PaymentRepository payments;
    private PaymentService paymentService;

    @BeforeEach
    void setUp() {
        payments = mock(PaymentRepository.class);
        paymentService = new PaymentService(payments);
    }

    @Test
    void getPayments() {
        when(payments.getByBankAccount("123-4567-89")).thenReturn(Arrays.asList(
                new Payment(100, "123-4567-89", "Freddy Kruger", "payment for: headline"))
        );
        assertThat(paymentService.getByBankAccount("123-4567-89")).containsExactly(
                new PaymentDto(100, "123-4567-89", "Freddy Kruger", "payment for: headline")
        );

    }
}

File: dotnet/niuz.application.test/services/PaymentServiceTest.cs
using System.Collections.Generic;
using FluentAssertions;
using niuz.application.dtos;
using niuz.application.entities;
using niuz.application.repositories;
using NSubstitute;
using Xunit;

namespace niuz.application.services
{
    public class PaymentServiceTest
    {
        private readonly IPaymentRepository payments;
        private readonly PaymentService paymentService;

        public PaymentServiceTest()
        {
            payments = Substitute.For<IPaymentRepository>();
            paymentService = new PaymentService(payments);
        }
        
        [Fact]
        public void GetPayments()
        {
            payments.GetByBankAccount("123-4567-89").Returns(new List<Payment> {new Payment(100, "123-4567-89", "Freddy Kruger", "payment for: headline")});
            paymentService.GetByBankAccount("123-4567-89").Should().ContainInOrder(
                new PaymentDto(100, "123-4567-89", "Freddy Kruger", "payment for: headline")
            );
        }
    }
}